Detectors/TPC/reconstruction/include/TPCReconstruction/KrBoxClusterFinder.h

```diff
@@ -187,35 +205,55 @@ class KrBoxClusterFinder
   int mSector = -1;                 ///< sector being processed in this instance
   std::unique_ptr<CalPad> mGainMap; ///< Gain map object
 
-  /// Maximum Map Dimensions
-  /// Here is room for improvements
-  static constexpr size_t MaxPads = 138;    ///< Size of the map in pad-direction
-  static constexpr size_t MaxRows = 152;    ///< Size of the map in row-direction
-  static constexpr size_t MaxTimes = 20000; ///< Size of the map in time-direction
-
   /// Values to define ROC boundaries
   static constexpr size_t MaxRowsIROC = 63;  ///< Amount of rows in IROC
   static constexpr size_t MaxRowsOROC1 = 34; ///< Amount of rows in OROC1
   static constexpr size_t MaxRowsOROC2 = 30; ///< Amount of rows in OROC2
   static constexpr size_t MaxRowsOROC3 = 25; ///< Amount of rows in OROC3
 
+  /// Vector of cluster objects
+  /// Used for returning results
   std::vector<o2::tpc::KrCluster> mClusters;
 
   /// Need an instance of Mapper to know position of pads
   const Mapper& mMapperInstance = o2::tpc::Mapper::instance();
 
   KrCluster mTempCluster; ///< Used to save the cluster data
 
-  /// Here the map is defined where all digits are temporarily stored
-  std::array<std::array<std::array<float, MaxPads>, MaxRows>, MaxTimes> mMapOfAllDigits{};
+  using TimeSliceSector = std::array<std::array<float, MaxPads>, MaxRows>;
+
+  /// A temporary timeslice that gets filled and can be added to the set of timeslices
+  TimeSliceSector mTempTimeSlice;
+
+  std::array<size_t, 36> mFirstDigit = {0};
```

Comment:
```suggestion
  size_t mFirstDigit = {0};
```
Since the KrBoxClusterFinder can only handle one sector, it does not make sense to create an array for all sectors here.